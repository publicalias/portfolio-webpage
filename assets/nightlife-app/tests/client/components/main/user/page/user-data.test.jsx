"use strict";

//local imports

const UserData = require("../../../../../../scripts/client/components/main/user/page/user-data");

const { newUserData } = require("../../../../../../schemas");
const { testWrapper } = require("../../../../test-helpers");

//global imports

const { initTestEvent, initTestSnapshot, withDataList } = require("redux/tests/client-tests");
const { reactTests } = require("redux/tests/react-tests");

//utilities

const { testShallow } = testWrapper(UserData);

//setup

beforeAll(reactTests.setup);
beforeEach(reactTests.inject(UserData));

//user data

describe("UserData", () => {

  const avatar = "https://www.example.com/avatar.jpg";

  const testData = withDataList(testShallow, [null, { userData: newUserData({ data: { avatar } }) }]);

  const testSnapshot = initTestSnapshot(testData);

  it("should match snapshot (default)", () => testSnapshot());

  it("should match snapshot (name)", () => testSnapshot(null, { userData: { name: "User A" } }));

  it("should call handleError on error", () => {

    const event = { target: { src: "" } };
    const placeholder = "https://via.placeholder.com/450x800?text=undefined";

    const testError = initTestEvent(testData, "error", event);

    return testError(".qa-error-image", [], () => {
      expect(event.target.src).toEqual(placeholder);
    });

  });

});
