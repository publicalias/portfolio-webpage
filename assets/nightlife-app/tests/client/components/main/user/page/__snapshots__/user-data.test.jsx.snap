// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserData should match snapshot (default) 1`] = `
<div>
  <h1
    className="u-align-left"
  >
    Anonymous
  </h1>
  <hr />
  <div
    className="c-user-avatar u-margin-full"
  >
    <img
      alt="User Photo"
      className="c-user-avatar__image qa-error-image"
      onError={[Function]}
      src="https://www.example.com/avatar.jpg"
    />
  </div>
  <p
    className="u-align-right u-margin-none"
  >
    0 Miles
  </p>
</div>
`;

exports[`UserData should match snapshot (name) 1`] = `
<div>
  <h1
    className="u-align-left"
  >
    User A
  </h1>
  <hr />
  <div
    className="c-user-avatar u-margin-full"
  >
    <img
      alt="User Photo"
      className="c-user-avatar__image qa-error-image"
      onError={[Function]}
      src="https://www.example.com/avatar.jpg"
    />
  </div>
  <p
    className="u-align-right u-margin-none"
  >
    0 Miles
  </p>
</div>
`;
